ballista/rust/scheduler/src/state/execution_graph.rs

```diff
@@ -485,8 +490,139 @@ impl ExecutionGraph {
         self.output_locations.clone()
     }
 
+    /// Reset running and completed stages on a given executor
```

Comment:
There is a lot of subtle logic here. I went through it the best I could and it seems ok but we should really have some test coverage for this.

Reply:
Yes, I already added two UTs to test the reset logic.

````
async fn test_reset_completed_stage()

async fn test_reset_resolved_stage()

````

Reply:
Sorry I missed those (it's a rather large PR :)). I think it might be worth adding some coverage for different graph topologies though (union/aggregation/coalesce). 

This may be a bit more involved but it may also be worthwhile to test the end-to-end flow from the grpc endpoint. There are quite a few moving parts here and given this is an exceptional situation it will be hard to track down issues that pop up in production.

Reply:
Sure, I can add UT to test from the grpc endpoint.